python/ray/util/tracing/tracing_helper.py

```diff
@@ -366,7 +366,8 @@ def _invocation_actor_class_remote_span(
         # If tracing feature flag is not on, perform a no-op
         if not is_tracing_enabled():
             if not self.__ray_metadata__.is_cross_language:
-                kwargs["_ray_trace_ctx"] = None
+                # Remove _ray_trace_ctx from kwargs if tracing disabled
+                kwargs.pop("_ray_trace_ctx", None)
```

Comment:
It seems inconsistent that trace ctx is not explicitly removed [here](https://github.com/ray-project/ray/pull/18926/files#diff-db449c12dfa112ed32d8fb4105900b8ca82ccdf7135f5a30d8d86e87041e27daR408) and at other places. Maybe leave a note that in future we add explicit removal at other callsites or turn this into assert "_ray_trace_ctx" not in kwargs?

Reply:
@ckw017 ^ can you do the same for other call sites and add the asserts?

Reply:
Will do in a follow-up